packages/block-editor/src/utils/object.js

```diff
@@ -93,24 +66,22 @@ function cloneObject( object ) {
  * @return {Object} Cloned object with the new value set.
  */
 export function setImmutably( object, path, value ) {
-	const normalizedPath = normalizePath( path );
-	const newObject = object ? cloneObject( object ) : {};
+	path = normalizePath( path );
+
+	if ( path.length === 0 ) {
+		throw new Error( 'Path cannot be empty' );
+	}
+
+	const shallowClone = Array.isArray( object )
+		? [ ...object ]
+		: { ...object };
+	const [ first, ...rest ] = path;
 
-	normalizedPath.reduce( ( acc, key, i ) => {
-		if ( acc[ key ] === undefined ) {
```

Comment:
Not sure if this is needed. The unit tests pass, so let's see if the e2e tests pass as well.